### PR TITLE
Use data types, remove anchors

### PR DESCRIPTION
**Pull request by juniorsysadmin**
- Only allow Boolean for $purge
- Only allow Boolean for $recurse
- Only allow Integer for $write_queue_limit_low
- Only allow Integer for $write_queue_limit_high
- Don't allow undef for $manage_service
- Replace anchors with contain
- Remove some unecessary tests

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -1,60 +1,51 @@
...
+  Boolean $internal_stats                              = $collectd::params::internal_stats,
+  Boolean $manage_package                              = $collectd::params::manage_package,
+  Boolean $manage_repo                                 = $collectd::params::manage_repo,
+  Optional[Collectd::Manifests::Init] $ci_package_repo = $collectd::params::ci_package_repo,
+  Boolean $manage_service                              = $collectd::params::manage_service,
+  String $minimum_version                              = $collectd::params::minimum_version,
+  String $package_ensure                               = $collectd::params::package_ensure,
+  Optional[Array] $package_install_options             = $collectd::params::package_install_options,
+  Stdlib::Fqdn $package_keyserver                      = $collectd::params::package_keyserver,
+  Variant[String[1],Array[String[1]]] $package_name    = $collectd::params::package_name,
```
Comment (juniorsysadmin) [inline manifests/init.pp]: #269 Appears to still affect recent Debian-variants

File: spec/classes/collectd_init_spec.rb [hunk 2]
```diff
@@ -9,20 +9,13 @@
 
       options = os_specific_options(facts)
       context 'with all defaults' do
-        it { is_expected.to contain_class('collectd') }
```
Comment (bastelfreak) [inline spec/classes/collectd_init_spec.rb]: why did you remove those two?

File: spec/classes/collectd_init_spec.rb [hunk 3]
```diff
@@ -9,20 +9,13 @@
...
         it { is_expected.to compile.with_all_deps }
-        it { is_expected.to contain_anchor('collectd::begin') }
-        it { is_expected.to contain_anchor('collectd::end') }
         it { is_expected.to contain_file('collectd.conf').without_content }
         it { is_expected.to contain_file('collectd.d').with_ensure('directory') }
         it { is_expected.to contain_file_line('include_conf_d').with_ensure('absent') }
         it { is_expected.to contain_file_line('include_conf_d_dot_conf').with_ensure('present') }
         it { is_expected.to contain_package(options[:package]).with_ensure('present') }
         it { is_expected.to contain_package(options[:package]).with_install_options(nil) }
-        it { is_expected.to contain_class('collectd::install').that_comes_before('Class[collectd::config]') }
```
Comment (bastelfreak) [inline spec/classes/collectd_init_spec.rb]: those checks should still be valid, since we use ->?

File: spec/classes/collectd_init_spec.rb (hunk 2 above)
Comment (juniorsysadmin) [reply spec/classes/collectd_init_spec.rb]: When inside `init.pp` we know that a class called collectd will always be there. I just didn't see any value to the test.
Same with params.

File: spec/classes/collectd_init_spec.rb (hunk 3 above)
Comment (juniorsysadmin) [reply spec/classes/collectd_init_spec.rb]: Valid, but again I don't think it is useful, as we don't have any conditional logic which will result in the order of classes changing.
